packages/babel-traverse/src/scope/index.js

```diff
@@ -9,59 +9,25 @@ import * as messages from "babel-messages";
 import Binding from "./binding";
 import globals from "globals";
 import * as t from "babel-types";
-
-//
-
-const CACHE_SINGLE_KEY = Symbol();
-const CACHE_MULTIPLE_KEY = Symbol();
+import { scope as scopeCache } from "../cache";
 
 /**
  * To avoid creating a new Scope instance for each traversal, we maintain a cache on the
  * node itself containing all scopes it has been associated with.
- *
- * We also optimise for the case of there being only a single scope associated with a node.
  */
 
 function getCache(node, parentScope, self) {
-  let singleCache = node[CACHE_SINGLE_KEY];
-
-  if (singleCache) {
-    // we've only ever associated one scope with this node so let's check it
-    if (matchesParent(singleCache, parentScope)) {
-      return singleCache;
-    }
-  } else if (!node[CACHE_MULTIPLE_KEY]) {
-    // no scope has ever been associated with this node
-    node[CACHE_SINGLE_KEY] = self;
-    return;
-  }
-
-  // looks like we have either a single scope association that was never matched or
-  // multiple assocations, let's find the right one!
-  return getCacheMultiple(node, parentScope, self, singleCache);
-}
+  let scopes: Array<Scope> = scopeCache.get(node) || [];
```

Comment:
We removed the rest of flow types in https://github.com/babel/babel/pull/3392

Do we want to add it in again now?